### PR TITLE
Add netstandard2.0 target to Grpc.Net.Client instrumentation

### PR DESCRIPTION
**Pull request by alanwest**
This is important for users consuming Grpc.Net.Client in .NET Framework applications.

There have been two previous attempts to introduce a `netstandard2.0` build of the Grpc.Net.Client instrumentation
* https://github.com/open-telemetry/opentelemetry-dotnet/pull/2007
* https://github.com/open-telemetry/opentelemetry-dotnet/pull/3062

We did not merge this previous PRs due to trickiness with testing the instrumentation's functionality for .NET Framework. Much of the current test suite spins up an in-process ASP.NET Core gRPC server. This is not possible for the `net462` target.

I've modified one of the existing tests to use a mock HttpClient, so that starting an in-process gRPC server is unnecessary.

The code that mocks up the HttpClient was fork-lifted from here: https://github.com/grpc/grpc-dotnet/tree/master/test/Shared.

This PR does not port the entire Grpc.Net.Client test suite to run on .NET Framework. Many of the tests test both the instrumentation of the gRPC call and the underlying HttpClient call. To my knowledge, testing the underlying HttpClient call requires making a real call since the mock HttpClient will not invoke the HttpClient instrumentation.

### CODE REVIEW

File: test/OpenTelemetry.Instrumentation.Grpc.Tests/GrpcTests.client.cs [hunk 1]
```diff
@@ -44,9 +46,17 @@ public partial class GrpcTests
...
+            var uri = new Uri($"{baseAddress}:1234");
             var uriHostNameType = Uri.CheckHostName(uri.Host);
 
+            var httpClient = ClientTestHelpers.CreateTestClient(async request =>
+            {
+                var streamContent = await ClientTestHelpers.CreateResponseContent(new HelloReply());
+                var response = ResponseUtils.CreateResponse(HttpStatusCode.OK, streamContent, grpcStatusCode: global::Grpc.Core.StatusCode.OK);
+                response.TrailingHeaders().Add("grpc-message", "value");
+                return response;
+            });
```
Comment (alanwest) [inline test/OpenTelemetry.Instrumentation.Grpc.Tests/GrpcTests.client.cs:58]: This test uses the pattern used to mock an http client in the `grpc-dotnet` project for testing the `Grpc.Net.Client`

For example:
https://github.com/grpc/grpc-dotnet/blob/eb813a798aa3c1009de65dc7e2c490106abd1401/test/Grpc.Net.Client.Tests/DiagnosticsTests.cs#L38-L45

File: test/OpenTelemetry.Instrumentation.Grpc.Tests/GrpcTests.client.cs [hunk 2]
```diff
@@ -104,6 +117,7 @@ public void GrpcClientCallsAreCollectedSuccessfully(string baseAddress, bool sho
             Assert.Equal(0, activity.GetTagValue(SemanticConventions.AttributeRpcGrpcStatusCode));
         }
 
+#if !NETFRAMEWORK
```
Comment (alanwest) [inline test/OpenTelemetry.Instrumentation.Grpc.Tests/GrpcTests.client.cs:120]: Many of the tests in this suite require an actual end-to-end call using a non-mocked HttpClient. To test the same functionality for .NET Framework, we would need to devise a more complicated integration test. Most likely involving spinning up Windows containers.

I'm proposing we acknowledge the short-comings of this test suite for .NET Framework and we ship a `netstandard2.0` version of the gRPC client instrumentation anyways.

File: test/OpenTelemetry.Instrumentation.Grpc.Tests/OpenTelemetry.Instrumentation.Grpc.Tests.csproj [hunk 3]
```diff
@@ -20,11 +21,15 @@
     <DotNetCliToolReference Include="dotnet-xunit" Version="$(DotNetXUnitCliVer)" />
 
     <PackageReference Include="Google.Protobuf" Version="$(GoogleProtobufPkgVer)" />
-    <PackageReference Include="Grpc.AspNetCore.Server" Version="$(GrpcAspNetCoreServerPkgVer)" />
+    <PackageReference Include="Grpc.AspNetCore.Server" Version="$(GrpcAspNetCoreServerPkgVer)" Condition="'$(TargetFramework)' != 'net462'" />
     <PackageReference Include="Grpc.Net.Client" Version="$(GrpcNetClientPkgVer)" />
     <PackageReference Include="Grpc.Tools" Version="$(GrpcToolsPkgVer)" PrivateAssets="All" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFramework)' == 'net462'">
```
Comment (reyang) [inline test/OpenTelemetry.Instrumentation.Grpc.Tests/OpenTelemetry.Instrumentation.Grpc.Tests.csproj:29]: curious if flatten this out would help to slightly increase the readability

```text
<Reference Include="System.Net.Http" Condition="'$(TargetFramework)' == 'net462'" />
```

**Review by reyang — APPROVED** [review]

File: test/OpenTelemetry.Instrumentation.Grpc.Tests/GrpcTests.client.cs (hunk 2 above)
Comment (cijothomas) [reply test/OpenTelemetry.Instrumentation.Grpc.Tests/GrpcTests.client.cs:120]: Agreed!

**Review by cijothomas — APPROVED** [review]

**Pull request merged**